ocs_ci/deployment/mce.py

```diff
@@ -0,0 +1,243 @@
+"""
+This module contains functionality required for mce installation.
+"""
+
+import logging
+import tempfile
+
+from ocs_ci.framework import config
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.utility import templating
+from ocs_ci.ocs import constants
+from ocs_ci.utility.utils import (
+    run_cmd,
+    exec_cmd,
+)
+from ocs_ci.ocs import exceptions
+from ocs_ci.ocs.resources.catalog_source import CatalogSource
+from ocs_ci.ocs import ocp
+from ocs_ci.utility.utils import get_running_ocp_version
+
+logger = logging.getLogger(__name__)
+
+
+class MCEInstaller(object):
+    """
+    mce Installer class for mce deployment
+    """
+
+    def __init__(self):
+        self.namespace = constants.MCE_NAMESPACE
+        self.ns_obj = ocp.OCP(kind=constants.NAMESPACES)
+        self.hypershift_override_image_cm = "hypershift-override-images-new"
+        self.multicluster_engine = ocp.OCP(
+            kind="MultiClusterEngine",
+            resource_name=constants.MULTICLUSTER_ENGINE,
+        )
+
+    def create_mce_catalog_source(self):
+        """
+        Creates a catalogsource for mce operator.
+
+        """
+        logger.info("Adding CatalogSource for MCE")
```

Comment:
we need to check if MCE catalogSource is not already installed. This way we insure, the Deployment can be rerun again from any installation stage. 
In opposite if we do not do this check, job will abort with error